src/libvirtApi/domain.js

```diff
@@ -999,43 +935,39 @@ export function domainSetMemory({
     return call(connectionName, objPath, 'org.libvirt.Domain', 'SetMemory', [memory, flags], { timeout, type: 'tu' });
 }
 
-export function domainSetMaxMemory({
+export async function domainSetMaxMemory({
     id: objPath,
     connectionName,
     maxMemory // in KiB
 }) {
-    return call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [0], { timeout, type: 'u' })
-            .then(domXml => {
-                const updatedXML = updateMaxMemory(domXml[0], maxMemory);
-                return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainDefineXML', [updatedXML], { timeout, type: 's' });
-            });
+    const [domXml] = await call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [0], { timeout, type: 'u' });
+    const updatedXML = updateMaxMemory(domXml, maxMemory);
+    await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainDefineXML', [updatedXML], { timeout, type: 's' });
 }
 
-export function domainSetOSFirmware({ connectionName, objPath, loaderType }) {
-    return call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' })
-            .then(domXml => {
-                const s = new XMLSerializer();
-                const doc = getDoc(domXml);
-                const domainElem = doc.firstElementChild;
+export async function domainSetOSFirmware({ connectionName, objPath, loaderType }) {
+    const [domXml] = await call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' });
+    const s = new XMLSerializer();
+    const doc = getDoc(domXml);
+    const domainElem = doc.firstElementChild;
 
-                if (!domainElem)
-                    return Promise.reject(new Error("setOSFirmware: domXML has no domain element"));
+    if (!domainElem)
+        throw new Error("setOSFirmware: domXML has no domain element");
 
-                const osElem = domainElem.getElementsByTagNameNS("", "os")[0];
-                const loaderElem = getSingleOptionalElem(osElem, "loader");
+    const osElem = domainElem.getElementsByTagNameNS("", "os")[0];
+    const loaderElem = getSingleOptionalElem(osElem, "loader");
 
-                if (loaderElem)
-                    loaderElem.remove();
+    if (loaderElem)
+        loaderElem.remove();
 
-                if (!loaderType)
-                    osElem.removeAttribute("firmware");
-                else
-                    osElem.setAttribute("firmware", loaderType);
+    if (!loaderType)
+        osElem.removeAttribute("firmware");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1397-20240122-164146-132eaa31-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)